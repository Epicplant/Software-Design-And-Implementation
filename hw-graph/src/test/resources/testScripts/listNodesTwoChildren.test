# Tests that we correctly have a list of 2 nodes+edges when ListChildren is called
# and the node has exactly 2 children

# Creates a new graph
CreateGraph newGraph

# Creates the first node
AddNode newGraph n1

# Creates the first nodes child
AddNode newGraph n2

# Creates the first nodes second child
AddNode newGraph n3

# Creates first node
AddEdge newGraph n1 n2 13

# Creates first node
AddEdge newGraph n1 n3 27

# Lists all of the children of this node
ListChildren newGraph n1
