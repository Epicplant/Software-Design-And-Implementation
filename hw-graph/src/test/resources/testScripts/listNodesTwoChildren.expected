# Tests that we correctly have a list of 2 nodes+edges when ListChildren is called
# and the node has exactly 2 children

# Creates a new graph
created graph newGraph

# Creates the first node
added node n1 to newGraph

# Creates the first nodes child
added node n2 to newGraph

# Creates the first nodes second child
added node n3 to newGraph

# Creates first node
added edge 13 from n1 to n2 in newGraph

# Creates first node
added edge 27 from n1 to n3 in newGraph

# Lists all of the children of this node
the children of n1 in newGraph are: n2(13) n3(27)
